### ISSUE TITLE
[bug] map `testID`-prop to `data-testid` on web

### ISSUE THREAD

**Issue by christianalares** [tamagui/tamagui#257]
**Describe the bug**
When adding `testID`-props the browser warns you:
```
React does not recognize the `testID` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `testid` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

This prop should resolve to `data-testid` in the browser. See [Button props example](https://necolas.github.io/react-native-web/docs/button/#props) on the react-native-web docs.

**To Reproduce**
Add a Tamagui element (like a `Button`) and add `testID="my-test-button"`. See the error in the console.

**System Info (please complete the following information):** 
  System:
    OS: macOS 12.4
    CPU: (10) arm64 Apple M1 Max
    Memory: 2.16 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 4.0.0-rc.13 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Firefox Developer Edition: 104.0
    Safari: 15.5
    Safari Technology Preview: 15.4
  npmPackages:
    @permanent/eslint-config: * => 0.0.0 
    eslint: 8.24.0 => 8.24.0 
    husky: 8.0.1 => 8.0.1 
    lint-staged: 13.0.3 => 13.0.3 
    prettier: ^2.7.1 => 2.7.1 
    turbo: 1.5.5 => 1.5.5 
    typescript: 4.8.4 => 4.8.4

**Comment by christianalares** [comment]
I have a temporary fix for now:

```
<Button
  // ...
  {...(process.env.TAMAGUI_TARGET === 'web' ? { 'data-testid': testID } : { testID })}
/>
```

**Comment by bang88** [comment]
@natew should we add this in `packages/core/src/helpers/assignNativePropsToWeb.tsx` ?

**Comment by natew** [comment]
What does react-native-web do here?

**Comment by natew** [comment]
Fixed in 0110b8030174136d0fb385baa60ef99f62424fda
